todaythediary/src/components/Essentials/Contact.js

```diff
@@ -0,0 +1,212 @@
+import React, { Component } from 'react'
+import {
+    Button, Form, FormGroup, Label, Input, Col, FormFeedback
+} from 'reactstrap';
+
+export default class Contact extends Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            firstname: '',
+            lastname: '',
+            telnum: '',
+            email: '',
+            agree: false,
+            contactType: 'Tel.',
+            message: '',
+            touched: {
+                firstname: false,
+                lastname: false,
+                telnum: false,
+                email: false
+            },
+            active: false
+        };
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleBlur = this.handleBlur.bind(this);
+    }
+
+
+    handleBlur = (field) => (evt) => {
+        this.setState({
+            touched: { ...this.state.touched, [field]: true }
+        });
+    }
+    validate(firstname, lastname, telnum, email) {
+        const errors = {
+            firstname: '',
+            lastname: '',
+            telnum: '',
+            email: '',
+            btn: false
+        };
+
+        if (this.state.touched.firstname && firstname.length < 3)
+            errors.firstname = 'First Name should be greater 3 characters';
+        else if (this.state.touched.firstname && firstname.length > 10)
+            errors.firstname = 'First Name should be less than 10 characters';
+
+        if (this.state.touched.lastname && lastname.length < 3)
+            errors.lastname = 'Last Name should be greater than 3 characters';
+        else if (this.state.touched.lastname && lastname.length > 10)
+            errors.lastname = 'Last Name should be less than 10 characters';
+
+        const reg = /^\d+$/;
+        if (this.state.touched.telnum && !reg.test(telnum))
+            errors.telnum = 'Tel. Number should contain only numbers ';
+        const validEmailRegex =
+            // eslint-disable-next-line no-useless-escape
+            RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+        if (this.state.touched.email) {
+            errors.email =
+                validEmailRegex.test(email)
+                    ? ''
+                    : 'Email is not valid!';
+        }
+        if ((!errors.firstname && !errors.lastname && !errors.telnum && !errors.email)) {
+            errors.btn = true
+        }
+
+        return errors;
+    }
+    handleInputChange(event) {
+        const target = event.target
+        const value = target.type === 'checkbox' ?
+            target.checked : target.value
+        const name = target.name
+
+        this.setState({
+            [name]: value,
+
+        })
```

Comment:
Consistent spacing, also are you using a formatter or Prettifier? If no, you can try it out